File: MKSMEVN.py
import sys
from sys import stdin
try:
    test=sys.stdin.readline()
    test=int(test)
    for z in range(test):
        n=sys.stdin.readline()
        n=int(n)
        a=list(map(int,input().strip().split()))
        sum=0
        for i in a:
            sum+=i
        if sum%2==0:
            print(0)
        else:
            ans=0
            for i in range(n):
                if(a[i]%2==0) and (a[i]+1)//2-1 <=0:
                    ans=1
                    break
            if ans==0:
                print(-1)
            else:
                print(1)
except EOFError:
    pass

File: MKSMEVN.cpp
#include <iostream>
using namespace std;

int main()
{
    int test;
    cin>>test;
    while(test--)
    {
        int n,i;
        cin>>n;
        int a[n];
        int sum=0;
        int answer;
        for(i=0;i<n;i++)
        {
            cin>>a[i];
            sum+=a[i];
        }
        if(sum%2==0)
        {
            cout<<0<<endl;
        }
        else
        {
            answer=0;
            for(i=0;i<n;i++)
            {
                if((a[i]%2==0) && ((a[i]+1)/2-1<=0))
                {
                    answer=1;
                    break;
                }
            }
            if(answer==0)
            {
                cout<<-1<<endl;
            }
            else
            {
                cout<<1<<endl;
            }
        }
    }
	return 0;
}
